File: solutions/csharp/log-levels/1/LogLevels.cs
static class LogLine
{
    public static string Message(string logLine)
    {
        string[] log = logLine.Split(" ");
        
        string s = "";

        foreach(String i in log)
            Console.WriteLine(i);

        int t = log.Length;

        for(int i=1;i<t;i++) {
            Console.WriteLine(i + " -> [" + log[i] + "] ... [" + log[i].Trim() + "]");

            if(log[i].Trim().Length > 0) {
                if(s.Length > 0) s += " ";
                s += log[i].Trim();
            }
        }
        
        return s;
    }

    public static string LogLevel(string logLine)
    {
        string[] s = logLine.Split(" ");

        int t = s[0].Trim().Length;
        
        return s[0].Trim().Substring(1,t-3).ToLower();
    }

    public static string Reformat(string logLine)
    {
        string msg = Message(logLine);
        string log = LogLevel(logLine);

        return msg + " (" + log + ")";
    }
}

File: solutions/java/log-levels/1/src/main/java/LogLevels.java
public class LogLevels {
    
    public static String message(String logLine) {
        int i=0;
        while(logLine.charAt(i) != ':'){
            i++;
        }

        int pos1 = logLine.length()-1;
        int pos2 = i;

        for(int j = i+2; j < logLine.length(); j++) {
            if(Character.isLetter(logLine.charAt(j))){
                pos1 = Math.min(pos1, j);
                pos2 = Math.max(pos2, j);
            }
        }

        return logLine.substring(pos1, pos2+1);
    }

    public static String logLevel(String logLine) {
        String h = "";
        for(int i=1; i<logLine.length() && logLine.charAt(i)!=']'; i++) {
            h += Character.toLowerCase(logLine.charAt(i));
        }
        return h;
    }

    public static String reformat(String logLine) {
        return message(logLine) + " (" + logLevel(logLine) + ")";
    }
}
